### ISSUE TITLE
Update from web-console 2.2.1

### ISSUE THREAD

**Issue by oslyak** [rails/web-console#204]
I have rails 4.2.6 app 

When I use web-console frozen to 2.2.1 everything works perfectly.
But when I try to upgrade web-console to any newest versions.
Even in development mode, when I made some mistake, I only see message: "500 Internal Server Error
If you are the administrator of this website, then please read this web application's log file and/or the web server's log file to find out what went wrong."

**Comment by gsamokovarov** [comment]
Did you do `bundle update` for the upgrade? `web-console` may be locked in your `Gemfile`. Can I see the `web-console` version in your `Gemfile.lock`?

**Comment by oslyak** [comment]
Yes, now I locked web-console on version 2.2.1, because I have troubles with new versions: Whe I install newest version, and make some mistake in code of my project, I only see message:  "500 Internal Server Error
If you are the administrator of this website, then please read this web application's log file and/or the web server's log file to find out what went wrong."

Without stack trace info. 

P.S. I have no troubles with installation process of new versions. Just remove Gemfile.lock and update version in Gemfile. Then "bundle update".

[Gemfile](http://paste.ubuntu.com/17636292/)
[Gemfile.lock](http://paste.ubuntu.com/17636293/)

**Comment by lebart** [comment]
Hello, 

I face the same bug, I have to revert back to version 3.1.1

**Comment by gsamokovarov** [comment]
Thanks for the reports, will look it up.

**Comment by gsamokovarov** [comment]
@lebart what ruby version are you using?

**Comment by lebart** [comment]
@gsamokovarov I'm using 2.2.3, should I test different versions ?

**Comment by gsamokovarov** [comment]
I'm asking because I dropped Ruby 1.9 support, so we can use newer Ruby API's and drop `binding_of_caller`. I'm not being able to reproduce the error on a clean rails 4.2.6 app, do you have any exception notifiers in your gem file? It may be that multiple gems are monkey patching `ActionDispatch::DebugExceptions` and they are changing it's interface.

**Comment by lebart** [comment]
I didn't see a gem that could interfere at this level but to be sure here are the gems I'm using:

[list of gems](http://paste.ubuntu.com/18043068/)

**Comment by oslyak** [comment]
I found the place!!!

In version 2.2.1 or lower everything works perfectly.

But if update to newest version and set application.rb

 config.action_view.raise_on_missing_translations = true

Then if you have mistake in your code even in development mode you will see:"500 Internal Server Error
If you are the administrator of this website, then please read this web application's log file and/or the web server's log file to find out what went wrong."

**Comment by wgrrrr** [comment]
I am seeing the same behaviour this issue describes. Here is a snippet of a backtrace which may help:
```
ArgumentError: wrong number of arguments (given 0, expected 1)
    from /bundle/gems/activesupport-4.2.7.1/lib/active_support/core_ext/kernel/reporting.rb:89:in `capture'
    from /bundle/gems/web-console-3.3.0/lib/web_console/view.rb:34:in `render'
    from /bundle/gems/web-console-3.3.0/lib/web_console/template.rb:20:in `render'
    from /bundle/gems/web-console-3.3.0/lib/web_console/middleware.rb:36:in `block in call'
    from /bundle/gems/web-console-3.3.0/lib/web_console/middleware.rb:18:in `catch'
    from /bundle/gems/web-console-3.3.0/lib/web_console/middleware.rb:18:in `call'
    ...
    ...
```

**Comment by mlt** [comment]
I have the same issue as @wgrrrr and same versions.

**Comment by gsamokovarov** [comment]
Can you give me a bit more context? I'll try to track this down.

**Comment by mlt** [comment]
@gsamokovarov Just a heads up... This is really odd. I was about to make a minimal reproducible example... and I failed at it. Brand new app with `rails _4.2.7_ new tmp` followed by bumping web-console to 3.3.0, works fine (at least on Windows). I'll try to narrow things down on my end.

**Comment by gsamokovarov** [comment]
@mit if you can get something reproducible, this will be of huge help!

**Comment by mlt** [comment]
@gsamokovarov I think I tracked it down to [this](https://gist.github.com/johnbintz/4684194#file-config_environments_development-rb-L3). I wanted live reloads and easier set up while being able to use foreman and see all logs dumped into console so I didn't notice when it broke down. So I guess in my case, it is not web-console to blame :-) Now I'm not even sure I actually need/want foreman.

**Comment by wgrrrr** [comment]
Interestingly, I have `config.logger = Logger.new(STDOUT)` in my `development.rb`. I don't use Foreman though, I have a Docker based containerized development environment that is orchestrated with docker-compose.

**Comment by gsamokovarov** [comment]
This do explains it. I'll handle the case in the next released version, for now, you can fix the error by setting an `ActiveSupport::Logger`  instead of the plain Ruby stdlib one.

**Comment by mlt** [comment]
I was not sure whether it was the problem @oslyak and @lebart had, but 2.2.1 does work alright with whatever logger. So I think everything is consistent. Thanks for the workaround!

**Comment by gsamokovarov** [comment]
`3.4.0` is released. I'm closing this thread as it's getting a bit general. If you have issues with `3.4.0`, please open more concrete issues.

**Comment by wflanagan** [comment]
It seems in 3.4.0, which fixes this problem, only compatible with Rails 5.0 upward?  What about "legacy" 4.2.7+ applications?

**Comment by Nowaker** [comment]
Theoretically, https://github.com/rails/web-console/commit/4ba9f5044f9322ea16f97e69b7167170c9c34522 should fix the problem introduced in commit https://github.com/rails/web-console/commit/19e60d4177fcea50fd920c68c500172640f706b7 released in version 3.2.0. Therefore, theoretically, this monkey patch should fix the problem:

```ruby
require 'web_console/view'

module WebConsole
  class View < ActionView::Base
    def render(*)
      if logger = WebConsole.logger and logger.respond_to?(:silence)
        WebConsole.logger.silence { super }
      else
        super
      end
    end
  end
end
```

But it doesn't. The silencing code, whatever it is, does not work when the logger in `config/environments/development.rb` is defined as `config.logger = Logger.new(STDOUT)`.

The newly introduced `def render(*)` in web-console 3.2.0 simply raises an exception on `super`. Adding a conditional and calling either `WebConsole.logger.silence { super }` or `super` doesn't do anything to fix the problem, since the problem is on the `super` statement. Therefore https://github.com/rails/web-console/commit/4ba9f5044f9322ea16f97e69b7167170c9c34522 was never a fix for the issue reported here.

Here's a proper monkey-patch that fixes web-console 3.2.0 on Rails 4.2.x.

```
require 'web_console/view'

module WebConsole
  class View < ActionView::Base
    remove_method :render
  end
end
```
